### PR TITLE
tesseract_ros 0.16.2 parity

### PR DESCRIPTION
**Pull request by rjoomen**
We have arrived at tesseract_ros 0.16.2 parity.
I also fixed some issues in the environment_monitor_interface and ported the tesseract_planning_server to ROS 2.

### CODE REVIEW

**Comment by rjoomen** [comment]
Done, both Humble and Foxy build. @marrts / @Levi-Armstrong could you review the PR?

Sorry for the messy commit history, I had some issues getting clang-format happy and foxy to build. clang-format still isn't happy, but that seems a config issue, as on my PC it does format the parts with preprocessor code properly. And the clang-tidy workflow needs an update, I'm looking into that.

**Comment by marip8** [comment]
FWIW the `.run-clang-format` script is using [`clang-format-10`](https://github.com/tesseract-robotics/tesseract_ros2/blob/8c2644e8c65c903bb69c8ee07530080da75ba8a0/.run-clang-format#LL2C187-L2C187) which may or may not be the version of `clang-format` installed on your machine. I've noticed differences in what `clang-format-8` chooses to format vs `clang-format-10`, so maybe that's part of the issue

**Comment by marrts** [comment]
I'm happy with this, but I'm not sure if @Levi-Armstrong or @marip8 have an opinion on the clang formatting being different. 

The only notable difference I see is now the switch-case statements have the `{` on the same line as the case statement rather than starting on a new line.

File: .clang-format [hunk 1]
```diff
@@ -54,7 +54,6 @@ BreakBeforeBraces: Custom
 
 # Control of individual brace wrapping cases
 BraceWrapping: {
-    AfterCaseLabel: 'true'
```
Comment (Levi-Armstrong) [inline .clang-format:57]: I would keep this the same to match all of Tesseract.

**Review by Levi-Armstrong — CHANGES_REQUESTED** [review]

File: .clang-format (hunk 1 above)
Comment (rjoomen) [reply .clang-format:57]: The tesseract_ros2 .clang-format is exactly the same as the one from tesseract, tesseract_planning and trajopt. So removing this line would make it different.

**Review by Levi-Armstrong — APPROVED** [review]

**Comment by Levi-Armstrong** [comment]
@marrts If your happy feel free to merge.

**Pull request merged**